tests/cpp/helpers.cc

```diff
@@ -227,15 +228,15 @@ std::unique_ptr<DMatrix> CreateSparsePageDMatrixWithRC(
     size_t j = 0;
     if (rem_cols > 0) {
        for (; j < std::min(static_cast<size_t>(rem_cols), cols_per_row); ++j) {
-         row_data << " " << (col_idx+j) << ":" << (col_idx+j+1)*10;
+         row_data << label(*gen) << " " << (col_idx+j) << ":" << (col_idx+j+1)*10*i;
```

Comment:
We implemented a dummy generator in helper, as the generator in std is not guaranteed to be reproducible on different compiler and different platform.

Reply:
I don't think we need to be reproducible across compilers/platforms, just need to be deterministic on multiple runs. Anyway, this is @sriramch's code, comments?

Reply:
that is correct. we just need the feature id's and values to be consistent when invoked multiple times for the same row/column configuration, when the deterministic flags is passed.